streams/src/test/java/org/apache/kafka/streams/processor/internals/GlobalStateManagerImplLogAndContinueTest.java

```diff
@@ -0,0 +1,129 @@
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import static java.util.Arrays.asList;
+import static org.apache.kafka.common.utils.Utils.mkEntry;
+import static org.apache.kafka.common.utils.Utils.mkMap;
+import static org.junit.Assert.assertEquals;
+
+public class GlobalStateManagerImplLogAndContinueTest extends GlobalStateManagerImplTest {
+
+    @Before
+    public void before() {
+        final Map<String, String> storeToTopic = new HashMap<>();
+
+        storeToTopic.put(storeName1, t1.topic());
+        storeToTopic.put(storeName2, t2.topic());
+        storeToTopic.put(storeName3, t3.topic());
+        storeToTopic.put(storeName4, t4.topic());
+        storeToTopic.put(storeName5, t5.topic());
+
+        store1 = new NoOpReadOnlyStore<>(storeName1, true);
+        store2 = new ConverterStore<>(storeName2, true);
+        store3 = new NoOpReadOnlyStore<>(storeName3);
+        store4 = new NoOpReadOnlyStore<>(storeName4);
+        store5 = new NoOpReadOnlyStore<>(storeName5, true);
+
+        final Deserializer<Long> longDeserializer = Serdes.Long().deserializer();
+        final MockSourceNode<Long, Long> source1 = new MockSourceNode<>(new String[]{t5.topic()}, longDeserializer, longDeserializer);
+
+        topology = new ProcessorTopology(Collections.<ProcessorNode>emptyList(),
+                mkMap(mkEntry(t5.topic(), source1)),
+                Collections.<String, SinkNode>emptyMap(),
+                Collections.<StateStore>emptyList(),
+                asList(store1, store2, store3, store4, store5),
+                storeToTopic,
+                Collections.emptySet());
+
+        streamsConfig = new StreamsConfig(new Properties() {
+            {
+                put(StreamsConfig.APPLICATION_ID_CONFIG, "appId");
+                put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "dummy:1234");
+                put(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory().getPath());
+                put(StreamsConfig.DEFAULT_DESERIALIZATION_EXCEPTION_HANDLER_CLASS_CONFIG, LogAndContinueExceptionHandler.class);
+            }
+        });
+        stateDirectory = new StateDirectory(streamsConfig, time, true);
+        consumer = new MockConsumer<>(OffsetResetStrategy.NONE);
+        stateManager = new GlobalStateManagerImpl(
+                new LogContext("test"),
+                topology,
+                consumer,
+                stateDirectory,
+                stateRestoreListener,
+                streamsConfig);
+        processorContext = new InternalMockProcessorContext(stateDirectory.globalStateDir(), streamsConfig);
+        stateManager.setGlobalProcessorContext(processorContext);
+        checkpointFile = new File(stateManager.baseDir(), ProcessorStateManager.CHECKPOINT_FILE_NAME);
+    }
+
+    @Test
+    public void shouldThrowStreamsExceptionWhenRestoringWithLogAndFailExceptionHandler() {
```

Comment:
The `GlobalStateManagerImplTest` already defines this test as well should this get removed here?

Reply:
@bbejeck This prevents the test from being executed, because it would fail because the exception would not be thrown in this setup. I could have prevented this by using the same name for the test but I thought a descriptive naming was more important.